src/components/timeline/Markers.js

```diff
@@ -27,6 +28,196 @@ export const MIN_MARKER_WIDTH = 0.3;
 
 type MarkerState = 'PRESSED' | 'HOVERED' | 'NONE';
 
+type MouseEventHandler = (SyntheticMouseEvent<HTMLCanvasElement>) => any;
+
+/**
+ * When adding properties to these props, please consider the comment above the component.
+ */
+type CanvasProps = {|
+  +rangeStart: Milliseconds,
+  +rangeEnd: Milliseconds,
+  +width: CssPixels,
+  +height: CssPixels,
+  +getMarker: MarkerIndex => Marker,
+  +markerIndexes: MarkerIndex[],
+  +hoveredItem: Marker | null,
+  +mouseDownItem: Marker | null,
+  +onMouseDown: MouseEventHandler,
+  +onMouseUp: MouseEventHandler,
+  +onMouseMove: MouseEventHandler,
+  +onMouseOut: MouseEventHandler,
+|};
+
+function _drawRoundedRect(
+  ctx: CanvasRenderingContext2D,
+  x: CssPixels,
+  y: CssPixels,
+  width: CssPixels,
+  height: CssPixels,
+  cornerSize: CssPixels
+) {
+  // Cut out c x c -sized squares in the corners.
+  const c = Math.min(width / 2, Math.min(height / 2, cornerSize));
+  const bottom = y + height;
+  ctx.fillRect(x + c, y, width - 2 * c, c);
+  ctx.fillRect(x, y + c, width, height - 2 * c);
+  ctx.fillRect(x + c, bottom - c, width - 2 * c, c);
+}
+
+/**
+ * This component controls the rendering of the canvas. Every render call through
+ * React triggers a new canvas render. Because of this, it's important to only pass
+ * in the props that are needed for the canvas draw call.
+ */
+class MarkersCanvas extends React.PureComponent<CanvasProps> {
+  _canvas: {| current: HTMLCanvasElement | null |} = React.createRef();
+  _requestedAnimationFrame: boolean = false;
```

Comment:
Strangely, this variable was present in the previous implementation but wasn't used. I aligned this with practices in other components.